### PR TITLE
Add filename parameter to ModelCheckpoint

### PR DESCRIPTION
**Pull request by wietsedv**
## What does this PR do?
Adds filename parameter to ModelCheckpoint. See issue #2583.

### CODE REVIEW

File: pytorch_lightning/callbacks/model_checkpoint.py [hunk 1]
```diff
@@ -93,8 +93,8 @@ class ModelCheckpoint(Callback):
 
     """
 
-    def __init__(self, filepath: Optional[str] = None, monitor: str = 'val_loss', verbose: bool = False,
-                 save_last: bool = False, save_top_k: int = 1, save_weights_only: bool = False,
+    def __init__(self, filepath: Optional[str] = None, filename: Optional[str] = None, monitor: str = 'val_loss',
```
Comment (Borda) [inline pytorch_lightning/callbacks/model_checkpoint.py:96]: Added argument in the middle can be breaking change as some users uses positioning arguments...

Comment (wietsedv) [reply pytorch_lightning/callbacks/model_checkpoint.py]: Thanks for the feedback. That's indeed a good point, so I fixed it.

File: pytorch_lightning/callbacks/model_checkpoint.py [hunk 2]
```diff
@@ -107,9 +107,11 @@ def __init__(self, filepath: Optional[str] = None, monitor: str = 'val_loss', ve
         self.monitor = monitor
         self.verbose = verbose
         if filepath is None:  # will be determined by trainer at runtime
-            self.dirpath, self.filename = None, None
+            self.dirpath, self.filename = None, filename
         else:
-            if os.path.isdir(filepath):
+            if filename is not None:
```
Comment (Borda) [inline pytorch_lightning/callbacks/model_checkpoint.py:112]:
```suggestion
            if filename:
```
an empty string is also not good

File: pytorch_lightning/callbacks/model_checkpoint.py [hunk 3]
```diff
@@ -95,7 +95,7 @@ class ModelCheckpoint(Callback):
 
     def __init__(self, filepath: Optional[str] = None, monitor: str = 'val_loss', verbose: bool = False,
                  save_last: bool = False, save_top_k: int = 1, save_weights_only: bool = False,
-                 mode: str = 'auto', period: int = 1, prefix: str = ''):
+                 mode: str = 'auto', period: int = 1, prefix: str = '', filename: Optional[str] = None):
```
Comment (Borda) [inline pytorch_lightning/callbacks/model_checkpoint.py:98]: missing docs for the new parameter

**Comment by rohitgr7** [comment]
I think this is a good feature, having separate dirpath and filename solves a lot of issues. I prefer it should be `dirpath` instead of `filepath`.

**Comment by Borda** [comment]
@wietsedv is this still needed after #3163 was merged? (then we can reopen it...)

**Pull request closed without merging**